AdRoll/baker
# feat: record validation in TOML

#### :question: What

Make it possible to declare Record validation from TOML. A new `Validation` key is added in the TOML, where the user can set a regex for each of the field names.

```
[Validation]
name = “^[a-z]*$“
timestamp = “^[0-9]*“
```

To avoid conflicts the validation could be omitted or set just one time either on the Component object or in the TOML.

#### :hammer: How to test

1. Add the Validation key on a Baker TOML
2. Make sure that no validation function is passed during the configuration
3. Records should be correctly validated and failed one throw away
#### :white_check_mark: Checklists

- [x] Is there unit/integration test coverage for all new and/or changed functionality added in this PR?
- [x] Have the changes in this PR been functionally tested?
- [ ] Have new components been added to the documentation website?
- [x] Has `make gofmt-write` been run on the code?
- [x] Has `make govet` been run on the code? Has the code been fixed accordingly to the output?
- [x] Have the changes been added to the [CHANGELOG.md](./CHANGELOG.md) file?
- [x] Have the steps in [CONTRIBUTING.md](./CONTRIBUTING.md) been followed to update a Go module?
